### PR TITLE
enh(ci): package lua dependencies using nfpm

### PR DESCRIPTION
**Pull request by hamzabessa**
## Description

package lua dependencies using nfpm

**Fixes** MON-33552

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/actions/package-nfpm/action.yml [hunk 1]
```diff
@@ -71,11 +71,19 @@ runs:
         export RPM_SIGNING_KEY_FILE="$(pwd)/key.gpg"
         export RPM_SIGNING_KEY_ID="$RPM_GPG_SIGNING_KEY_ID"
         export NFPM_RPM_PASSPHRASE="$RPM_GPG_SIGNING_PASSPHRASE"
-
-        sed -i "s/@luaver@/$luaver/g" ./${{ inputs.nfpm_file_pattern }}
-        sed -i "s/@COMMIT_HASH@/${{ inputs.commit_hash }}/g" ./${{ inputs.nfpm_file_pattern }}
-        nfpm package --config ./${{ inputs.nfpm_file_pattern }} --packager ${{ inputs.package_extension }}
-
+        echo "Current directory: $(pwd)"
+        version=`date '+%Y%m%d'`
```
Comment (kduret) [inline .github/actions/package-nfpm/action.yml:75]: use major_version and minor_version inputs

File: .github/actions/package-nfpm/action.yml [hunk 2]
```diff
@@ -71,11 +71,19 @@ runs:
...
+        for FILE in ${{ inputs.nfpm_file_pattern }}; do
+          DIRNAME=$(dirname $FILE)
+          BASENAME=$(basename $FILE)
+          cd $DIRNAME
+          sed -i "s/@luaver@/$luaver/g" $BASENAME
+          sed -i "s/@VERSION@/$version/g" $BASENAME
```
Comment (kduret) [inline .github/actions/package-nfpm/action.yml:81]: use environment variable

File: .github/docker/Dockerfile.packaging-alma8 [hunk 3]
```diff
@@ -15,4 +15,4 @@ dnf clean all
 
 EOF
 
-WORKDIR /src
+WORKDIR /src
```
Comment (kduret) [inline .github/docker/Dockerfile.packaging-alma8:18]: revert change

File: .github/docker/Dockerfile.packaging-alma9 [hunk 4]
```diff
@@ -15,4 +15,4 @@ dnf clean all
 
 EOF
 
-WORKDIR /src
+WORKDIR /src
```
Comment (kduret) [inline .github/docker/Dockerfile.packaging-alma9:18]: revert change

File: .github/docker/Dockerfile.packaging-bullseye [hunk 5]
```diff
@@ -16,4 +16,4 @@ apt-get clean
 
 EOF
 
-WORKDIR /src
+WORKDIR /src
```
Comment (kduret) [inline .github/docker/Dockerfile.packaging-bullseye:19]: revert change

File: .github/workflows/stream-connectors-dependencies.yml [hunk 6]
```diff
@@ -9,51 +9,63 @@ on:
   pull_request:
     paths:
       - dependencies/**
+      - '.github/workflows/stream-connectors-dependencies.yml'
```
Comment (kduret) [inline .github/workflows/stream-connectors-dependencies.yml:12]:
```suggestion
```

File: .github/workflows/stream-connectors-dependencies.yml [hunk 7]
```diff
@@ -9,51 +9,63 @@ on:
...
   push:
     branches:
       - develop
       - master
     paths:
       - dependencies/**
+      - '.github/workflows/stream-connectors-dependencies.yml'
```
Comment (kduret) [inline .github/workflows/stream-connectors-dependencies.yml:19]:
```suggestion
```

File: .github/workflows/stream-connectors-dependencies.yml [hunk 8]
```diff
@@ -77,20 +88,24 @@ jobs:
           cat version.txt
 
           mkdir ${{ matrix.lib }}
+          if  [ "${{ matrix.package_extension }}" = "rpm" ]; then
           cp -rp /usr/share/lua/`lua -e "print(string.sub(_VERSION, 5))"`/luatz/* ${{ matrix.lib }}/
-          tar czf ~/rpmbuild/SOURCES/${{ matrix.lib }}.tar.gz ${{ matrix.lib }}
+          else
+          cp -rp /usr/local/share/lua/`lua -e "print(string.sub(_VERSION, 5))"`/luatz/* ${{ matrix.lib }}/
+          fi
```
Comment (kduret) [inline .github/workflows/stream-connectors-dependencies.yml:95]: bad indentation

File: .github/actions/package-nfpm/action.yml [hunk 9]
```diff
@@ -85,7 +90,7 @@ runs:
         key: ${{ inputs.cache_key }}
 
     # Update if condition to true to get packages as artifacts
-    - if: ${{ false }}
+    - if: ${{ true }}
```
Comment (tuntoja) [inline .github/actions/package-nfpm/action.yml:93]: will this be permanent, won t we reach the artifacts storage limit ?

File: dependencies/lua-cffi/packaging/lua-cffi.yaml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+homepage: "https://www.centreon.com"
+license: "Apache-2.0"
+
+contents:
+  - src: "../lua-cffi/cffi.so"
+    dst: "/usr/lib64/lua/@luaver@/cffi.so"
+    packager: rpm
+
+  - src: "../lua-cffi/cffi.so"
+    dst: "/usr/lib64/lua/5.3/cffi.so"
```
Comment (kduret) [inline dependencies/lua-cffi/packaging/lua-cffi.yaml:23]:
```suggestion
    dst: "/usr/lib/x86_64-linux-gnu/lua/5.3/cffi.so"
```

File: dependencies/lua-cffi/packaging/lua-cffi.yaml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+contents:
+  - src: "../lua-cffi/cffi.so"
+    dst: "/usr/lib64/lua/@luaver@/cffi.so"
+    packager: rpm
+
+  - src: "../lua-cffi/cffi.so"
+    dst: "/usr/lib/x86_64-linux-gnu/lua/5.3/cffi.so"
+    packager: deb
+  - src: "../lua-cffi/cffi.so"
+    dst: "/usr/lib64/lua/5.4/cffi.so"
```
Comment (kduret) [inline dependencies/lua-cffi/packaging/lua-cffi.yaml:26]:
```suggestion
    dst: "/usr/lib/x86_64-linux-gnu/lua/5.4/cffi.so"
```

File: .github/actions/package-nfpm/action.yml [hunk 12]
```diff
@@ -68,26 +65,34 @@ runs:
           exit 1
         fi
 
+        export VERSION="${{ inputs.version }}"
         export RPM_SIGNING_KEY_FILE="$(pwd)/key.gpg"
         export RPM_SIGNING_KEY_ID="$RPM_GPG_SIGNING_KEY_ID"
         export NFPM_RPM_PASSPHRASE="$RPM_GPG_SIGNING_PASSPHRASE"
+        echo "Current directory: $(pwd)"
```
Comment (kduret) [inline .github/actions/package-nfpm/action.yml:72]:
```suggestion
```

File: .github/actions/package-nfpm/action.yml [hunk 13]
```diff
@@ -68,26 +65,33 @@ runs:
...
     - name: Cache packages
-      uses: actions/cache/save@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
       with:
         path: ./*.${{ inputs.package_extension }}
         key: ${{ inputs.cache_key }}
 
     # Update if condition to true to get packages as artifacts
-    - if: ${{ false }}
+    - if: ${{ true }}
```
Comment (kduret) [inline .github/actions/package-nfpm/action.yml:92]:
```suggestion
    - if: ${{ false }}
```

File: .github/docker/Dockerfile.packaging-bullseye [hunk 14]
```diff
@@ -5,15 +5,13 @@ FROM ${REGISTRY_URL}/debian:bullseye
 RUN bash -e <<EOF
 
 apt-get update
-apt-get install -y debmake pbuilder aptitude zstd ca-certificates lua5.4
+apt-get install -y git debmake pbuilder aptitude zstd ca-certificates luarocks lua5.4 liblua5.4-dev
```
Comment (kduret) [inline .github/docker/Dockerfile.packaging-bullseye:8]:
```suggestion
apt-get install -y git debmake pbuilder aptitude zstd ca-certificates luarocks lua5.3 liblua5.3-dev
```

File: .github/actions/package-nfpm/action.yml [hunk 15]
```diff
@@ -68,26 +65,33 @@ runs:
           exit 1
         fi
 
+        export VERSION="${{ inputs.version }}"
```
Comment (kduret) [inline .github/actions/package-nfpm/action.yml:68]:
```suggestion
```

File: .github/actions/package-nfpm/action.yml [hunk 16]
```diff
@@ -68,26 +65,33 @@ runs:
...
-        sed -i "s/@luaver@/$luaver/g" ./${{ inputs.nfpm_file_pattern }}
-        sed -i "s/@COMMIT_HASH@/${{ inputs.commit_hash }}/g" ./${{ inputs.nfpm_file_pattern }}
-        nfpm package --config ./${{ inputs.nfpm_file_pattern }} --packager ${{ inputs.package_extension }}
-
+        for FILE in ${{ inputs.nfpm_file_pattern }}; do
+          DIRNAME=$(dirname $FILE)
+          BASENAME=$(basename $FILE)
+          cd $DIRNAME
+          sed -i "s/@luaver@/$luaver/g" $BASENAME
+          sed -i "s/@COMMIT_HASH@/${{ inputs.commit_hash }}/g" $BASENAME
```
Comment (kduret) [inline .github/actions/package-nfpm/action.yml:78]:
```suggestion
          sed -i "s/@luaver@/$luaver/g" $BASENAME
          sed -i "s/@VERSION@/${{ inputs.version }}/g" $BASENAME
          sed -i "s/@COMMIT_HASH@/${{ inputs.commit_hash }}/g" $BASENAME
```

**Review by paul-oureib — DISMISSED** [review]

**Review by bouda1 — APPROVED** [review]

**Review by paul-oureib — APPROVED** [review]

**Pull request merged**